records/2021-07-16.scalability.database-bottleneck.md

```diff
@@ -0,0 +1,135 @@
+# Database performance issues
+
+Date: 2021-07-16
+
+## Status
+
+**DRAFT**
+
+## Context
+
+Because Notify suffers from regular daily incident that originates from its
+database when clients blasts notification, we need to review some design aspect
+of the stack and bring more control to specific elements.
+
+In the past months, the current architecture, environment and configuration have
+been tweaked to achieve a certain balance that could take on the expected load.
+To list a few of these tweaks:
+
+* the number of maximum pool connection in individual application instance,
+* the AWS instance classes,
+* the resources requests and limits of Kubernetes containers,
+* the horizontal pod autoscaling.
+
+These serve well the application but there were already signs of the
+application's limit: we rarely if ever meet our SLAs and we had regular daily
+incidents that foretold of the difficulties ahead.
+
+This document propose various options to address thorny issues that our
+database is having.
```

Comment:
where can we find the current database diagram? a link in this ADR will give more context

Reply:
PR is incomplete and diagrams to come soon.